src/v2/System/Router/buildClientAppContext.ts

```diff
@@ -27,6 +27,7 @@ export const buildClientAppContext = (
       mediator,
       user: sd.CURRENT_USER,
       featureFlags: sd.FEATURE_FLAGS,
+      userPreferences: sd.USER_PREFERENCES,
```

Comment:
**question**: maybe it would be better if we store it in the `user` field? Something like this

```javascript
{
  analytics: {
    contextPageOwnerSlug: pageSlug,
    contextPageOwnerType: pageType,
  },
  mediator,
  user: {
    ...sd.CURRENT_USER,
    preferences: sd.USER_PREFERENCES,
  },
  featureFlags: sd.FEATURE_FLAGS,
  ...context,
}
```

Reply:
This is a good question. One thing about the `user` object is that it maps 1:1 with the model. If we were to extend it it would almost be better if we were to make the change in Gravity so that no additional field or object merging is needed (if I'm understanding correctly)

Reply:
> If we were to extend it it would almost be better if we were to make the change in Gravity so that no additional field or object merging is needed

we can query additional fields via `userKeys` [here](https://github.com/artsy/force/blob/b1603f4b74c73dea950dcf4869a5294c6e8c7f0d/src/middleware.ts#L198) (see example below). But the main problem is that the `length_unit_preference` will contain the value that was relevant at the time of executing [this request](https://github.com/artsy/force/blob/b1603f4b74c73dea950dcf4869a5294c6e8c7f0d/src/lib/passport/lib/app/token_login.js#L43) and as far as I understood, the values specified in `userKeys` will not be automatically updated until the next user authorization (or are there moments that I missed?)

```javascript
app.use(
  artsyPassport({
    ...
    userKeys: [
      "collector_level",
      "default_profile_id",
      "email",
      "has_partner_access",
      "id",
      "lab_features",
      "name",
      "paddle_number",
      "phone",
      "roles",
      "type",
      "length_unit_preference", // 👈 query new field
    ],
  })
)
```